Jarvis.Framework/ProjectionEngine/Atomic/AtomicReadModelSignatureFixer.cs

```diff
@@ -156,20 +156,19 @@ private async Task InnerFix()
                             fixCheckpoint = elementToFix.ProjectedPosition;
                             count++;
                             _logger.DebugFormat("Fixing readmodel {0}", elementToFix.Id);
-                            T fixedRm = default(T);
+
+                            //To avoid problem with exception we will create a readmodel without any event then catchup
+                            T fixedRm = _atomicReadModelFactory.Create<T>(elementToFix.Id);

```

Comment:
Questo perchè con il factory si crea un readmodel vergine all'ultima versione (e quindi serializzabile/deserializzabile in ogni caso) rispetto al default? 
Il catchup partendo da zero non tiene conto di snapshot, giusto? 
Nel caso ok, ma in passato si parlava di rimuovere vecchi eventi o archiviarli, in quel caso sarebbe da rivedere questa parte?

Reply:
I readmodel NON hanno snapshot, era una different branch che non ho mai finito perchè avevo trovato un paio di cose strane.
Fattto il catchup per non dovrere modificare l'infrastruttura base del projector, quello che si usava prima che era il "full projection" se internamente da errore, da eccezione e non torna nulla, per cui se di 10 eventi ne ha proiettati 3 comunque ti torna null.

Invece a me serve trappare l'eccezione ed avere il readmodel con i 3 eventi proiettati, gli metto faulted = true cosi il vecchio non va a fare casino.

Reply:
Quindi il ragionamento è: ti creo fuori il readmodel a zero, poi ti faccio riproiettare tutto con il catchup il quale lo prende a versione 0 per cui in realtà proietta tutto :).